### PR TITLE
Add Bosnian locale

### PR DESCRIPTION
**Pull request by BranislavLazic**
This PR adds Bosnian locale and resolves this issue https://github.com/date-fns/date-fns/issues/2435.

### CODE REVIEW

File: src/locale/bs/_lib/formatRelative/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  yesterday: "'juče u' p",
+  today: "'danas u' p",
+  tomorrow: "'sutra u' p",
+  nextWeek: function (date) {
+    var day = date.getUTCDay()
+
+    switch (day) {
+      case 0:
+        return "'sledeće nedjelje u' p"
```
Comment (sake92) [inline src/locale/bs/_lib/formatRelative/index.js:24]: I think that `sljedeće` is correct one here.

Comment (sake92) [reply src/locale/bs/_lib/formatRelative/index.js:24]: Also `sedmica` instead of `nedjelja`.

Comment (BranislavLazic) [reply src/locale/bs/_lib/formatRelative/index.js]: Fixed. Thank you!

File: src/locale/bs/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+var locale = {
+  code: 'bs',
+  formatDistance: formatDistance,
+  formatLong: formatLong,
+  formatRelative: formatRelative,
+  localize: localize,
+  match: match,
+  options: {
+    weekStartsOn: 1 /* Monday */,
+    firstWeekContainsDate: 1,
```
Comment (tan75) [inline src/locale/bs/index.js:24]: hi @BranislavLazic 
Could you please provide a reference that confirms the value of 1 of `firstWeekContainsDate`? 
According to Wikipedia, it is 4 for the majority of European countries. 
Do you possibly know a source that we can refer to?

**Review by tan75 — CHANGES_REQUESTED** [review]

File: src/locale/bs/index.js (hunk 2 above)
Comment (BranislavLazic) [reply src/locale/bs/index.js:24]: Seems like you are right. Also, some of them share the same flaw. E.g. sr-Latn (which was my reference locale) or hr.

**Comment by tan75** [comment]
@BranislavLazic 
could you please re-generate snapshots by running `yarn locale-snapshots` and commit the results?

**Review by tan75 — CHANGES_REQUESTED** [review]
Please run `yarn locale-snapshots` and commit the results.

**Review by tan75 — APPROVED** [review]
Much appreciated! 👍  👍  
I will schedule this for our next release given the go-ahead by @kossnocorp

**Pull request merged**